OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -193,69 +187,31 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
     }
 
     private fun initIAPObservers() {
-        iapViewModel.executeResponse.observe(
-            viewLifecycleOwner,
-            NonNullObserver {
-                if (iapViewModel.upgradeMode.isSilentMode()) {
-                    iapDialog.showNewExperienceAlertDialog(this, { _, _ ->
-                        iapViewModel.showFullScreenLoader(true)
-                    }, { _, _ ->
-                        resetPurchase()
-                    })
-                }
-            })
-
-        iapViewModel.showFullscreenLoaderDialog.observe(
-            viewLifecycleOwner,
-            NonNullObserver { canShowLoader ->
-                if (canShowLoader) {
-                    fullscreenLoader?.show(childFragmentManager, FullscreenLoaderDialogFragment.TAG)
-                    iapViewModel.showFullScreenLoader(false)
-                }
-            })
-
-        iapViewModel.refreshCourseData.observe(
-            viewLifecycleOwner,
-            NonNullObserver { refreshCourse ->
-                if (refreshCourse) {
-                    refreshOnPurchase = true
-                    courseViewModel.fetchEnrolledCourses(
-                        type = CoursesRequestType.LIVE,
-                        showProgress = false
-                    )
-                    iapViewModel.refreshCourseData(false)
-                }
-            })
-
-        iapViewModel.purchaseFlowComplete.observe(
-            viewLifecycleOwner,
-            NonNullObserver { isPurchaseCompleted ->
-                if (isPurchaseCompleted) {
-                    if (fullscreenLoader?.isAdded == true) fullscreenLoader?.dismiss()
-                    iapAnalytics.trackIAPEvent(Analytics.Events.IAP_COURSE_UPGRADE_SUCCESS)
-                    iapAnalytics.trackIAPEvent(Analytics.Events.IAP_UNLOCK_UPGRADED_CONTENT_TIME)
-                    iapAnalytics.trackIAPEvent(Analytics.Events.IAP_UNLOCK_UPGRADED_CONTENT_REFRESH_TIME)
-                    SnackbarErrorNotification(binding.root).showError(R.string.purchase_success_message)
-                    iapViewModel.resetPurchase(false)
-                }
-            })
-
-        iapViewModel.errorMessage.observe(viewLifecycleOwner, NonNullObserver { errorMsg ->
-            if (iapViewModel.upgradeMode.isNormalMode()) {
-                return@NonNullObserver
+        iapViewModel.refreshCourseData.observe(viewLifecycleOwner, NonNullObserver { iapFlowData ->
+            iapDialog.showNewExperienceAlertDialog(this, { _, _ ->
+                showFullscreenLoader(iapFlowData)
+            }, { _, _ -> })
+        })
+        iapViewModel.errorMessage.observe(viewLifecycleOwner, NonNullObserver { errorMessage ->
```

Comment:
Add a new line. 

Please check for new lines and spaces in other Observers as well.